Coding standards guideline topic (#5)
Alp4ga/SQA_Assignment_1: Handbook.md

__Links:__
### 3.1 Topics introduction
### 3.2 Important guidelines
* Easy team integration
* Increased code quality efficiency and easy for maintaining
* Reduce code complexity
* Reduce development cost

* Naming Convention
* File and folder Naming and Organization

> * Formatiting and Indentation

Reviewer: typo
Formatiting -> Formatting

```suggestion
* Formatting and Indentation
```

* Commenting and Documenting
* Classes and Functions
* Testing
### 3.3 Topics
The developer's world has too much coding standards. It is important in a software development to select the better one. This coding standards depends on the language used i.e., in C++ you will not use the same coding standard than if the language will be in Java or in PHP. It will also depend on the company. You will have three best options, the easiest option is to follow a coding standard which already exists, another way will be to follow a coding standard and make small change because it does not feat with your company rules. The last option will be to create your own coding standards.

> The most difficults for a developeur is to follow or work with another developer who does not use the same coding standards. During the development you must create new folders and files. It is important to use the same named and structured format to lose less time as possible when you are looking for a specific file. i.e an asset like a picture or a font.

Reviewer: typo
developeur -> developer

I also took time to change a sentence that didn't sound right IMO

```suggestion
The most difficults for a developer is to follow or work with another developer who does not use the same coding standards. During the development you must create new folders and files. It is important to use the same naming convention and folder structure to save as much time as possible when you are looking for a specific file. i.e an asset like a picture or a font.
```

> Read code from someone else can be hard when the nameming conventions is not the same. For exemple all class names must be in PascalCase and the variables must be in camelCase. If you use an opposed standards convention for the nameming the other person could not understand really fast what your code is supposed to do. Our team will lose time and the development cost will be increased.

Reviewer: typo 2 times
namening -> naming

```suggestion
Read code from someone else can be hard when the naming conventions is not the same. For exemple all class names must be in PascalCase and the variables must be in camelCase. If you use an opposed standards convention for the naming the other person could not understand really fast what your code is supposed to do. Our team will lose time and the development cost will be increased.
```

> When you are working in team on the same project an important part is the testing. Two co-working could be working on the same features and without testing the other co-worker could broke your part without knowing what he is doing. The testing has this role to inssure no one is broking your work with additional code.

Reviewer: typo
inssure -> insure
co-working -> co-workers ?

grammar
is broking -> broke

```suggestion
When you are working in team on the same project an important part is the testing. Two co-workers could be working on the same feature and without testing the other co-worker could break your part without knowing what he is doing. The testing has this role to insure no one broke your work with additional code.
```

> All developeurs do not have the same abilities or the same competencies. It is important to commenting and documenting. It will make it easy for the other co-working to understand what is appening in your class, your function, or your package. Without commenting the developer must read and understant the whole concept to work with it. Again, it will increase the time so the development cost.

Reviewer: typo
developeurs -> developers
co-working -> co-worker
appening -> happening

grammar
commenting -> comment
documenting -> document

```suggestion
All developers do not have the same abilities or the same skills. It is important to comment and document. It will make it easy for the other co-workers to understand what is happening in your class, your function or your package. Without comments, the developer must read and understand the whole concept to work with it. Again, it will increase the time so the development cost.
```

> The identation is also important to increase the readability of the code. A bad identation will render code reaaly hard to follow.

Reviewer: typo
identation -> indentation

```suggestion
The indentation is also important to increase the code readability. A bad indentation will make the code really hard to understand.
```

> There are no bad coding standards, the important his to follow the same coding standards for the all software development. The coding standard must be defined at the start of the development. Everyone who is working on the project must also follow this coding standard to assure the good work of the software develpment.

Reviewer: typo
his -> is
assure -> insure

```suggestion
There are no bad coding standards, the important is to follow the same coding standards for the all software development. The coding standard must be defined at the start of the development. Everyone who is working on the project must also follow this coding standard to insure the good work of the software develpment.
```

Reviewer: and I have to say that this sentence doesn't make much sense IMO
`the important is to follow the same coding standards for the all software development.`

__Links:__
### 3.1 Topics introduction
### 3.2 Important guidelines
* Easy team integration
* Increased code quality efficiency and easy for maintaining
* Reduce code complexity
* Reduce development cost

* Naming Convention
* File and folder Naming and Organization
* Formatting and Indentation
* Commenting and Documenting
* Classes and Functions
* Testing
* Code Linters
### 3.3 Topics

> The developer's world has too much coding standards. It is important in a software development to select the better one. This coding standards depends on the language used i.e., in C++ you will not use the same coding standard than if the language will be in Java or in PHP. It will also depend on the company. You will have three best options, the easiest option is to follow a coding standard which already exists, another way will be to follow a coding standard and make small change because it does not feat with your company rules. The last option will be to create your own coding standards.

Reviewer 2: ```suggestion
The developer's world has too much coding standards. It is important in a software development to select the better one. Those coding standards depends on the language used, i.e. in C++ you would not use the same coding standard as with Java or PHP. It will also depend on the company. You will have three best options, the easiest option is to follow a coding standard which already exists, another way will be to follow a coding standard and make small change because it does not feat with your company rules. The last option will be to create your own coding standards.
```

> The most difficults for a developer is to follow or work with another developer who does not use the same coding standards. During the development you must create new folders and files. It is important to use the same naming convention and folder structure to save as much time as possible when you are looking for a specific file. i.e an asset like a picture or a font.

Reviewer 2: ```suggestion
The most difficult for a developer is to follow or work with another developer who does not use the same coding standards. During the development you must create new folders and files. It is important to use the same naming convention and folder structure to save as much time as possible when you are looking for a specific file. i.e an asset like a picture or a font.
```

> Read code from someone else can be hard when the naming conventions is not the same. For exemple all class names must be in PascalCase and the variables must be in camelCase. If you use an opposed standards convention for the naming the other person could not understand really fast what your code is supposed to do. Our team will lose time and the development cost will be increased.

Reviewer 2: ```suggestion
Read code from someone else can be hard when the naming conventions are not the same. For example all class names must be in PascalCase and the variables must be in camelCase. If you use an opposed standards convention for the naming, the other person could not understand what your code is supposed to do really fast. Our team will lose time and the development cost will be increased.
```

> When you are working in team on the same project an important part is the testing. Two co-workers could be working on the same feature and without testing the other co-worker could break your part without knowing what he is doing. The testing has this role to insure no one broke your work with additional code.

Reviewer 2: ```suggestion
When you are working in a team on the same project an important part is the testing. Two co-workers could be working on the same feature, and without testing, the other co-worker could break your part without knowing what he is doing. Testing has this role to insure no one broke your work with additional code.
```

> All developers do not have the same abilities or the same skills. It is important to comment and document. It will make it easy for the other co-workers to understand what is happening in your class, your function or your package. Without comments, the developer must read and understand the whole concept to work with it. Again, it will increase the time so the development cost.

Reviewer 2: ```suggestion
All developers does not have the same abilities or the same skills. It is important to comment and document. It will make it easy for the other co-workers to understand what is happening in your class, your function or your package. Without comments, the developer must read and understand the whole concept to work with it. Again, it will increase the time, so the development cost.
```

The indentation is also important to increase the code readability. A bad indentation will make the code really hard to understand.

> The last important point on Coding Standards is to use tools to help you to increse your code quality. There we are talking about Code Linters; It is a tool which analyze your source code looking for problems. It will fewer errors in production, it will make readable, maintainable, and more consistent code. With this tool you will avoid syntax errors which is the most basic and more vital type of checks that a linter can provide.

Reviewer 2: ```suggestion
The last important point on Coding Standards is to use tools to help you to increase your code quality. Here we are talking about Code Linters; It is a tool which analyze your source code looking for problems. It will reduce errors in production, but also make readable, maintainable and more consistent code. With this tool you will avoid syntax errors which is the most basic and more vital type of checks that a linter can provide.
```

> There are no bad coding standards, the important is to follow the same coding standards for the all software development. The coding standard must be defined at the start of the development. Everyone who is working on the project must also follow this coding standard to insure the good work of the software develpment.

Reviewer 2: ```suggestion
There are no bad coding standards, the important is to follow the same coding standards during all the software development. The coding standard must be defined at the start of the development. Everyone who is working on the project must also follow this coding standard to insure the good work of the software development.
```
